File: 1935-maximum-number-of-words-you-can-type/1935-maximum-number-of-words-you-can-type.cpp
class Solution {
public:
    int canBeTypedWords(string text, string brokenLetters) {
        unordered_set<char> unavailable;

        for (const char& c : brokenLetters) unavailable.insert(c);

        vector<string> words;
        istringstream ss(text);
        string w;

        while (ss >> w) words.push_back(w);

        int ans = 0;
        for (const string& word : words){
            bool available = true;
            for (const char& c : word){
                if (unavailable.find(c) != unavailable.end()) {
                    available = false;
                    break;
                }
            }

            if (available) ++ans;
        }
        
        return ans;
    }
};

File: 1935-maximum-number-of-words-you-can-type/1935-maximum-number-of-words-you-can-type.py
class Solution:
    def canBeTypedWords(self, text: str, brokenLetters: str) -> int:
        unavailable, words = set(brokenLetters), [word for word in text.split()]

        ans = 0
        for word in words:
            available = True
            for c in word:
                if c in unavailable:
                    available = False
                    break
            if available:
                ans += 1
        
        return ans